File: NutriSmart/app/src/main/java/com/example/nutrismart/data/UserSession.kt
package com.example.nutrismart.data

object UserSession {
    // In-memory cache for the current session (backed by SessionManager on disk)
    var currentUserId: Long = -1L
    var token: String = ""
    var isGoogleUser: Boolean = false

    fun clear() {
        currentUserId = -1L
        token = ""
        isGoogleUser = false
    }
}

File: NutriSmartIOS/data/UserSession.swift
//
//  UserSession.swift
//  NutriSmartIOS
//

import Foundation

class UserSession {
    static let shared = UserSession()
    
    var currentUserId: Int = -1
    var token: String = ""
    var isGoogleUser: Bool = false

    private init() {}

    func clear() {
        currentUserId = -1
        token = ""
        isGoogleUser = false
    }
}
